pyuvdata/uvdata/uvh5.py

```diff
@@ -524,8 +530,12 @@ def _read_header(
 
         # get frequency information
         self.freq_array = header["freq_array"][:, :]
-        self.channel_width = float(header["channel_width"][()])
         self.spw_array = header["spw_array"][:]
+        # We've added a new keyword, that did not exist before, so
+        if self.flex_spw:
+            self.channel_width = header["channel_width"][:]
+        else:
+            self.channel_width = float(header["channel_width"][()])
```

Comment:
Yup -- we'd still need the split handling if `flex_spw` isn't found as a keyword in the header (since that implies that `channel_width` is a solitary float), so some version of this would get moved up into line 448.